### PR TITLE
mcobbett rest cps adapter

### PR DESCRIPTION
**Pull request by techcobweb**
## Why change ?

See 
[Local test run to be able to use remote CPS #1813](https://github.com/galasa-dev/projectmanagement/issues/1813)

## What changed ?
- Fixing javadoc warnings so we can see errors and warnings easier
- Move common extension code into a separate bundle
- Move shared mock classes to a separate bundle/project
- initial code cut of the cps-rest extension
- secure namespace in CPS should not be used by local test runs
- a few CPS configuration properties should not be used by local test runs.
  - eg: Where the DSS and Creds are stored.

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by eamansour — APPROVED** [review]
Looks good

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by eamansour — APPROVED** [review]

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: galasa-extensions-parent/dev.galasa.extensions.common/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Bundle-License: https://www.eclipse.org/legal/epl-2.0
+Export-Package: dev.galasa.extensions.common.*
+Import-Package: \
+dev.galasa,\
+dev.galasa.framework.spi,\
+javax.net.ssl,\
+javax.security.cert,\
+org.apache.commons.logging,\
+org.apache.http.impl.client
+Embed-Transitive: true
```
Comment (techcobweb) [inline galasa-extensions-parent/dev.galasa.extensions.common/bnd.bnd:13]: Embedding these dependencies many times is bad. Separate item added to the story to sort this out.

**Pull request merged**